Invalid Json Tests fail despite being correctly invalid

### Environment
* Python version: 3.8.5
* schema-enforcer version:  develop-branch

<!-- What did you expect to happen? -->
### Expected Behavior
Schema Invalid tests pass by failing the same.

<!-- What happened instead? -->
### Observed Behavior
Schema Invalid tests fail.

This is because the jsonschema __results_ arent being cleared between running the valid tests and the invalid tests.
This leads to _results_sorted_ and _expected_results_sorted_ being different, as the previous pass of the valid test is still being included in the results.

**results_sorted:**
`[{'instance_location': '/home/philipp/Documents/public/schema-enforcer/schema/tests/test/valid', 'instance_name': 'full_implementation.json', 'instance_type': 'TEST', 'result': 'PASS', 'schema_id': 'schemas/test'}, {'absolute_path': [...], 'message': "123 is not of type 'string'", 'result': 'FAIL', 'schema_id': 'schemas/test'}]`

**expected_results_sorted:**
`[{'absolute_path': [...], 'message': "123 is not of type 'string'", 'result': 'FAIL', 'schema_id': 'schemas/test'}]`

<!--
    Describe in detail the exact steps that someone else can take to reproduce
    this bug using the current release.
-->
### Steps to Reproduce
1. Have Json File to be validated.
2. Have a valid test in the subfolder tests/*/valid/
3. Have invalid tests in the tests/*/invalid/ subfolder.
4. Generate the expected results for the invalid tests.
5. Run schema-enforcer schema --check

I first saw this on production data, also the same on the following testdata:
```
schema
├── schemas
│   └── test.yml
└── tests
    └── test
        ├── invalid
        │   ├── invalid_type1
        │   │   ├── data.json
        │   │   └── results.yml
        │   └── invalid_type2
        │       ├── data.json
        │       └── results.yml
        └── valid
            └── full_implementation.json
```

**cat schema/schemas/test.yml** 
```
---
$schema: "http://json-schema.org/draft-07/schema#"
$id: "schemas/test"
description: "NFTables Firewall Configuration schema."
type: "object"
properties:
  firewall:
    type: "object"
    uniqueItems: true
    additionalProperties: false
    required:
      - "rule"
      - "variables"
    properties:
      rule:
        type: "object"
        properties:
          bool:
            type: "boolean"
          Text:
            type: "string"
          dict:
            type: "object"
      variables:
        type: "object"
```

**cat schema/tests/test/invalid/invalid_type1/data.json** 
```
{
    "firewall": {
        "rule": {
            "bool": "true",
            "Text": "text",
            "dict": {
            }
        },
        "variables": {
            "Text": "text",
            "array": [
            ]
        }
    }
}
```

**cat schema/tests/test/invalid/invalid_type2/data.json** 
```
{
    "firewall": {
        "rule": {
            "bool": true,
            "Text": 123,
            "dict": {
            }
        },
        "variables": {
            "Text": "text",
            "array": {
            }
        }
    }
}
```


**cat test.json**
```
{
    "firewall": {
        "rule": {
            "bool": true,
            "Text": "text",
            "dict": {
            }
        },
        "variables": {
            "Text": "text",
            "array": [
            ]
        }
    }
}
```



The following patch fixes the issue for me:
```
diff --git a/schema_enforcer/schemas/manager.py b/schema_enforcer/schemas/manager.py
index e34b4fd..9658e80 100644
--- a/schema_enforcer/schemas/manager.py
+++ b/schema_enforcer/schemas/manager.py
@@ -201,6 +201,7 @@ class SchemaManager:
         test_dirs = next(os.walk(invalid_test_dir))[1]
 
         results = []
+        schema.clear_results()
         for test_dir in test_dirs:
             data_file = find_file(os.path.join(invalid_test_dir, test_dir, "data"))
             expected_results_file = find_file(os.path.join(invalid_test_dir, test_dir, "results"))
```

Thanks for submitting this issue.

I believe I was able to recreate it with what you've supplied, both with `--generate-invalid` and `--check`.

I get this error when running `--check`:

```shell
FAIL | [ERROR] Invalid test results do not match expected test results from /home/user/repos/testing/sebug/schema/tests/test/invalid/invalid_type2/results.yml [PROPERTY]
FAIL | [ERROR] Invalid test results do not match expected test results from /home/user/repos/testing/sebug/schema/tests/test/invalid/invalid_type1/results.yml [PROPERTY]
```

Is that similar to what you saw in your environment?

I also confirmed your proposed fix. We'll get a PR submitted and add some tests to catch this behavior in the future.

--- a/schema_enforcer/schemas/manager.py
+++ b/schema_enforcer/schemas/manager.py
@@ -202,6 +202,7 @@ class SchemaManager:
 
         results = []
         for test_dir in test_dirs:
+            schema.clear_results()
             data_file = find_file(os.path.join(invalid_test_dir, test_dir, "data"))
             expected_results_file = find_file(os.path.join(invalid_test_dir, test_dir, "results"))
 
@@ -263,7 +264,7 @@ class SchemaManager:
 
         # For each test, load the data file, test the data against the schema and save the results
         for test_dir in test_dirs:
-
+            schema.clear_results()
             data_file = find_file(os.path.join(invalid_test_dir, test_dir, "data"))
 
             if not data_file:

--- a/tests/fixtures/test_manager/invalid_generate/schema/schemas/test.yml
+++ b/tests/fixtures/test_manager/invalid_generate/schema/schemas/test.yml
@@ -1,0 +1,25 @@
+---
+$schema: "http://json-schema.org/draft-07/schema#"
+$id: "schemas/test"
+description: "NFTables Firewall Configuration schema."
+type: "object"
+properties:
+  firewall:
+    type: "object"
+    uniqueItems: true
+    additionalProperties: false
+    required:
+      - "rule"
+      - "variables"
+    properties:
+      rule:
+        type: "object"
+        properties:
+          bool:
+            type: "boolean"
+          Text:
+            type: "string"
+          dict:
+            type: "object"
+      variables:
+        type: "object"

--- a/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type1/data.json
+++ b/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type1/data.json
@@ -1,0 +1,13 @@
+{
+    "firewall": {
+        "rule": {
+            "bool": "true",
+            "Text": "text",
+            "dict": {}
+        },
+        "variables": {
+            "Text": "text",
+            "array": []
+        }
+    }
+}

--- a/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type1/exp_results.yml
+++ b/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type1/exp_results.yml
@@ -1,0 +1,9 @@
+---
+results:
+  - result: "FAIL"
+    schema_id: "schemas/test"
+    absolute_path:
+      - "firewall"
+      - "rule"
+      - "bool"
+    message: "'true' is not of type 'boolean'"

--- a/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type2/data.json
+++ b/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type2/data.json
@@ -1,0 +1,13 @@
+{
+    "firewall": {
+        "rule": {
+            "bool": true,
+            "Text": 123,
+            "dict": {}
+        },
+        "variables": {
+            "Text": "text",
+            "array": {}
+        }
+    }
+}

--- a/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type2/exp_results.yml
+++ b/tests/fixtures/test_manager/invalid_generate/schema/tests/test/invalid/invalid_type2/exp_results.yml
@@ -1,0 +1,9 @@
+---
+results:
+  - result: "FAIL"
+    schema_id: "schemas/test"
+    absolute_path:
+      - "firewall"
+      - "rule"
+      - "Text"
+    message: "123 is not of type 'string'"

--- a/tests/fixtures/test_manager/invalid_generate/schema/tests/test/valid/test.json
+++ b/tests/fixtures/test_manager/invalid_generate/schema/tests/test/valid/test.json
@@ -1,0 +1,13 @@
+{
+    "firewall": {
+        "rule": {
+            "bool": true,
+            "Text": "text",
+            "dict": {}
+        },
+        "variables": {
+            "Text": "text",
+            "array": []
+        }
+    }
+}

--- a/tests/test_schemas_schema_manager.py
+++ b/tests/test_schemas_schema_manager.py
@@ -51,3 +51,34 @@ def test_invalid():
         schema_manager = SchemaManager(config=Settings(**config))  # noqa pylint: disable=unused-variable
         expected_error = """Invalid JSONschema file: invalid.yml - ["'bla bla bla bla' is not of type 'object', 'boolean'", "'integer' is not of type 'object', 'boolean'"]"""
         assert expected_error in str(e)
+
+
+def test_generate_invalid(capsys):
+    """ Test validates that generate_invalid_test_expected generates the correct data. """
+    config = {
+        "main_directory": os.path.join(FIXTURE_DIR, "test_manager", "invalid_generate", "schema"),
+    }
+    schema_id = "schemas/test"
+
+    schema_manager = SchemaManager(config=Settings(**config))
+    schema_manager.generate_invalid_tests_expected(schema_id)
+
+    invalid_dir = os.path.join(config["main_directory"], "tests", "test", "invalid")
+    invalid_tests = ["invalid_type1", "invalid_type2"]
+
+    for test in invalid_tests:
+        test_dir = os.path.join(invalid_dir, test)
+        with open(os.path.join(test_dir, "exp_results.yml")) as exp_file:
+            expected = exp_file.read()
+        with open(os.path.join(test_dir, "results.yml")) as gen_file:
+            generated = gen_file.read()
+        assert expected == generated
+
+    test_schema = schema_manager.schemas.get(schema_id)
+    # Clear results as these would not be carried over into subsequent run of --check
+    test_schema.clear_results()
+    # Ignore earlier output
+    capsys.readouterr()
+    schema_manager.test_schemas()
+    captured = capsys.readouterr()
+    assert "ALL SCHEMAS ARE VALID" in captured.out

